### PR TITLE
AEA-1613 fix: validate aea_version when loading configuration (#2519)

### LINKED ISSUE #2519: MAM: adding an outdated agent doesn't show a proper error
**Describe the bug**
MAM: adding an outdated agent doesn't show a proper error.
It seems like config validation goes before version validation.

**To Reproduce**
- Add an agent `Aeolist/simple_weather_buyer:0.2.0` with AEA v1.0.1 via MultiAgentManager
- See the "Invalid config" error

**Expected behavior**
Show an error that agent version doesn't match the current AEA version.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: any
 - AEA Version 1.0.1

```
[1] Traceback (most recent call last):
[1]   File "/Users/oleg/Documents/work/agents-gui/app/back/views.py", line 129, in fetch_agent
[1]     request.app.manager.add_agent(

[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/manager/manager.py", line 482, in add_agent
[1]     self.add_project(public_id, local, remote, restore)
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/manager/manager.py", line 394, in add_proje
ct
[1]     project = Project.load(
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/manager/project.py", line 126, in load
[1]     do_fetch(ctx, public_id, is_local, is_remote, target_dir=target_dir)
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/cli/fetch.py", line 102, in do_fetch
[1]     fetch_agent(ctx, public_id, alias=alias, target_dir=target_dir)
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/cli/utils/decorators.py", line 221, in wrap
per
[1]     raise e
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/cli/utils/decorators.py", line 218, in wrap
per
[1]     return func(context, *args, **kwargs)
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/cli/registry/fetch.py", line 86, in fetch_a
gent
[1]     try_to_load_agent_config(ctx)
[1]   File "/Users/oleg/.local/share/virtualenvs/back-yqq0ceD0/lib/python3.8/site-packages/aea/cli/utils/config.py", line 96, in try_to_lo
ad_agent_config
[1]     raise click.ClickException(
[1] click.exceptions.ClickException: Agent configuration file 'aea-config.yaml' is invalid: `ExtraPropertiesError: properties not expected
: registry_path`. Please check the documentation.
```

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Validate aea_version when loading configuration.

That is, before validating the loaded YAML file against the schema, first check that `aea_version` is consistent with the current `aea` version being used.

It also introduces the `skip_aea_validation` flag in the `ConfigLoader` constructor (and factory methods/functions). This is useful to switch the behaviour of configuration loading in terms of `aea_version` validation.

As pointed out by @DavidMinarsch , sometime we don't want to validate the `aea_version`, e.g. we can fetch packages even if the current AEA version is incompatible with the fetched package; but we want such validation for the MAM.

## Fixes

#2519

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Upgrade tests fail, we probably have to skip validation for packages which are about to be upgraded. We should also make it possible to fetch packages which are outdated, this should just not be possible on AEA manager, but should be possible on fetch. So I'd say the new validation logic added needs to be by default off and only on when we use it in manager.

File: aea/manager/project.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def load(
         cli_verbosity: str = "INFO",
         registry_path: str = DEFAULT_REGISTRY_NAME,
         skip_consistency_check: bool = False,
+        skip_aea_validation: bool = True,
```
Comment (DavidMinarsch) [inline aea/manager/project.py:101]: why skip during load?

Comment (marcofavorito) [reply aea/manager/project.py:101]: will put it to `False`

Comment (marcofavorito) [reply aea/manager/project.py:101]:
```suggestion
        skip_aea_validation: bool = False,
```

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**